src/megaapi_impl.cpp

```diff
@@ -14216,73 +14278,155 @@ void MegaApiImpl::getua_result(byte* data, unsigned len, attr_t type)
 
 void MegaApiImpl::getua_result(TLVstore *tlv, attr_t)
 {
-    if(requestMap.find(client->restag) == requestMap.end()) return;
-    MegaRequestPrivate* request = requestMap.at(client->restag);
-    if(!request || (request->getType() != MegaRequest::TYPE_GET_ATTR_USER)) return;
+    error e = API_OK;
+    MegaRequestPrivate* request = NULL;
+    auto it = requestMap.find(client->restag);
+    if (it == requestMap.end() || !(request = it->second)
+           || (request->getType() != MegaRequest::TYPE_GET_ATTR_USER && request->getType() != MegaRequest::TYPE_SET_ATTR_USER)
+           || (request->getType() == MegaRequest::TYPE_SET_ATTR_USER && request->getParamType() != MegaApi::USER_ATTR_ALIAS)) return;
 
     if (tlv)
     {
-        // TLV data usually includes byte arrays with zeros in the middle, so values
-        // must be converted into Base64 strings to avoid problems
-        MegaStringMap *stringMap = new MegaStringMapPrivate(tlv->getMap(), true);
-        request->setMegaStringMap(stringMap);
-
-        switch (request->getParamType())
+        if (request->getType() == MegaRequest::TYPE_SET_ATTR_USER)
```

Comment:
This `if` should finish in a `return`, since you don't want to finish the request at this stage, but when the `putua()` finishes. That way you can also avoid moving the existing block into an `else` (and generating an inconvenient diff).